lib/rubocop/cop/lint/parentheses_as_grouped_expression.rb

```diff
@@ -48,17 +48,22 @@ def valid_context?(node)
             return true
           end
 
-          node.operator_method? || node.setter_method? || grouped_parentheses?(node)
+          node.operator_method? || node.setter_method? || chained_calls?(node) ||
+            logical_operator?(node)
         end
 
         def first_argument_starts_with_left_parenthesis?(node)
           node.first_argument.source.start_with?('(')
         end
 
-        def grouped_parentheses?(node)
+        def chained_calls?(node)
           first_argument = node.first_argument
+          first_argument.send_type? && (node.children.last&.children&.count || 0) > 1
+        end
 
-          first_argument.send_type? && first_argument.receiver&.begin_type?
+        def logical_operator?(node)
+          first_argument = node.first_argument
+          (first_argument.and_type? || first_argument.or_type?) && first_argument.logical_operator?
```

Comment:
Can you use `operator_keyword?`?

```suggestion
          first_argument.operator_keyword? && first_argument.logical_operator?
```

And the test passes even if `&& first_argument.logical_operator?` will be removed. Is this condition necessary?

Reply:
Thank you for suggesting this, it seems much simpler 👍 🙇